lib/Models/Form.php

```diff
@@ -120,4 +124,29 @@ static function groupBy($orm, $group = null) {
     return $orm->whereNull('deleted_at');
   }
 
+  static function getDefaultSuccessMessage() {
+    $settings = new SettingsController;
+    if ($settings->get('signup_confirmation.enabled')) {
+      return WPFunctions::get()->__(self::MESSAGE_WHEN_CONFIRMATION_ENABLED, 'mailpoet');
+    }
+    return WPFunctions::get()->__(self::MESSAGE_WHEN_CONFIRMATION_DISABLED, 'mailpoet');
+  }
+
+  static function updateSuccessMessages() {
+    $right_message = self::getDefaultSuccessMessage();
+    $wrong_message = (
+      $right_message === self::MESSAGE_WHEN_CONFIRMATION_ENABLED
+      ? self::MESSAGE_WHEN_CONFIRMATION_DISABLED
+      : self::MESSAGE_WHEN_CONFIRMATION_ENABLED
+    );
+    $forms = self::findMany();
```

Comment:
The form settings are stored in database as a serialized array. That's why I can't use an `UPDATE` statement, because I need to unserialize the settings first to find the message and decide if I need to update it or not. I was using `asArray` because it unserializes the settings, I will replace it.